lib/fizzy/span.hpp

```diff
@@ -0,0 +1,46 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2019-2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <type_traits>
+#include <vector>
+
+namespace fizzy
+{
+/// The span describes an object that can refer to a contiguous sequence of objects with the first
+/// element of the sequence at position zero.
+///
+/// This is minimal implementation of C++20's std::span:
+/// https://en.cppreference.com/w/cpp/container/span
+/// Only `const T` is supported.
+template <typename T, typename = typename std::enable_if<std::is_const_v<T>>::type>
+struct span
```

Comment:
any reason why it's a `struct` and not `class`?

Reply:
As it's not a standard method, I think it should be called `Span` like our other classes like `Stack` and `OperandStask`

Reply:
This is from C++20: https://en.cppreference.com/w/cpp/container/span.
But it is a class so I will change to class.

Reply:
It's not, because it's not in `std`.

But it's just a nitpick, I don't care much.

Reply:
I will leave it as is to be a drop-in replacement for standardized "container".